### PR TITLE
Handle cancellation tokens correctly in 'fix all occurences'

### PR DESCRIPTION
**Pull request by shyamnamboodiripad**
Use the correct cancellation token in 'fix all occurences' and also bail on cancellation in more places in the 'fix all occurences' code path.

### CODE REVIEW

**Comment by shyamnamboodiripad** [comment]
@mavasani @tmeschter @heejaechang @srivatsn @jmarolf Could you please take a look?

**Comment by mavasani** [comment]
:+1:

**Comment by shyamnamboodiripad** [comment]
@heejaechang  @srivatsn I need one more thumbs up for this. Could one of you take a look (should be quick)

File: src/Features/Core/CodeFixes/FixAllOccurrences/FixAllCodeAction.cs [hunk 1]
```diff
@@ -43,10 +43,12 @@ public override string Title
 
         protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             FixAllLogger.LogContext(_fixAllContext, IsInternalCodeFixProvider(_fixAllContext.CodeFixProvider));
 
             var service = _fixAllContext.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
-            return await service.GetFixAllOperationsAsync(_fixAllProvider, _fixAllContext).ConfigureAwait(false);
+            // Use the new cancellation token instead of the stale one present inside _fixAllContext.
+            return await service.GetFixAllOperationsAsync(_fixAllProvider, _fixAllContext.WithCancellationToken(cancellationToken)).ConfigureAwait(false);
```
Comment (shyamnamboodiripad) [inline src/Features/Core/CodeFixes/FixAllOccurrences/FixAllCodeAction.cs]: The primary fix is here. We were using the wrong cancellation token before.

**Comment by srivatsn** [comment]
:+1:

**Comment by heejaechang** [comment]
:+1:

**Pull request merged**